### DISCUSSION TITLE
blog/cloud/2024-12-02-cloud-architecture-with-aws-rds-elasticache/

### DISCUSSION THREAD

**Discussion by giscus** [mingdodev/mingdodev.github.io#22] [Announcements]
# blog/cloud/2024-12-02-cloud-architecture-with-aws-rds-elasticache/

moonshot

https://mingdodev.github.io/blog/cloud/2024-12-02-cloud-architecture-with-aws-rds-elasticache/

**Comment by mingdodev** [comment]
왜 레디스를 쓴 것이었나요?!
redis와 memcached 그리고 valkey까지 비교해보도록 하세요

**Comment by ajroot5685** [comment]
Elastic cache는 어떤 목적으로 사용되는건가요?
노드 복제를 통해 가용성을 유지해야할만큼 아키텍처에서 중요한 역할을 수행하나요?

**Reply by mingdodev** [reply]
회원 가입 시 이메일 인증을 위한 키값을 3분간 key-value 형태로 저장할 때 사용됩니다. 따라서 분산해서 저장할 만큼 데이터가 크지는 않다고 판단되어 클러스터링 모드는 사용하지 않았고, 대신 프라이머리 노드가 다운되었을 때 자동 failover 정도는 해줄 수 있게 노드 복제를 채택했습니다.

사실 설계 목적이 안 써본 기능 써보며 `AWS와 친해지기`가 우선이라, 서비스 사용자가 없는 현 상황에서는 오버엔지니어링이 맞긴 합니다. 특히 가용성을 누려보고 싶어서 이것저것 해봤는데 정확히 파악해주셨네요 😮😝
